pkg/pquerier/query_integration_test.go

```diff
@@ -3286,6 +3286,460 @@ func (suite *testQuerySuite) TestSelectDataframeDaownsampleMetricsHaveBigGaps()
 	suite.Require().Equal(1, dataFrameCount, "series count didn't match expected")
 }
 
+func (suite *testQuerySuite) TestClientWindowedAggregationWindowBiggerThanStep() {
+	adapter, err := tsdb.NewV3ioAdapter(suite.v3ioConfig, nil, nil)
+	if err != nil {
+		suite.T().Fatalf("failed to create v3io adapter. reason: %s", err)
+	}
+
+	labels1 := utils.LabelsFromStringList("os", "linux")
+	numberOfEvents := 10
+
+	var ingestedData []tsdbtest.DataPoint
+
+	for i := 0; i < numberOfEvents; i++ {
+		ingestedData = append(ingestedData, tsdbtest.DataPoint{Time: suite.basicQueryTime + int64(i)*tsdbtest.MinuteInMillis, Value: 10 * float64(i)})
+	}
+
+	testParams := tsdbtest.NewTestParams(suite.T(),
+		tsdbtest.TestOption{
+			Key: tsdbtest.OptTimeSeries,
+			Value: tsdbtest.TimeSeries{tsdbtest.Metric{
+				Name:   "cpu",
+				Labels: labels1,
+				Data:   ingestedData},
+			}})
+	tsdbtest.InsertData(suite.T(), testParams)
+
+	expected := map[string][]tsdbtest.DataPoint{"sum": {
+		{Time: suite.basicQueryTime, Value: 0},
+		{Time: suite.basicQueryTime + 5*tsdbtest.MinuteInMillis, Value: 150},
+		{Time: suite.basicQueryTime + 10*tsdbtest.MinuteInMillis, Value: 390},
+	}}
+
+	querierV2, err := adapter.QuerierV2()
+	if err != nil {
+		suite.T().Fatalf("Failed to create querier v2, err: %v", err)
+	}
+
+	params := &pquerier.SelectParams{Name: "cpu",
+		Functions:         "sum",
+		Step:              5 * tsdbtest.MinuteInMillis,
+		AggregationWindow: 6 * tsdbtest.MinuteInMillis,
+		From:              suite.basicQueryTime,
+		To:                suite.basicQueryTime + 10*tsdbtest.MinuteInMillis}
+	set, err := querierV2.Select(params)
+	if err != nil {
+		suite.T().Fatalf("Failed to exeute query, err: %v", err)
+	}
+
+	var seriesCount int
+	for set.Next() {
+		seriesCount++
+		iter := set.At().Iterator()
+
+		data, err := tsdbtest.IteratorToSlice(iter)
+		agg := set.At().Labels().Get(aggregate.AggregateLabel)
+		if err != nil {
+			suite.T().Fatal(err)
+		}
+
+		assert.Equal(suite.T(), expected[agg], data, "queried data does not match expected")
+	}
+
+	assert.Equal(suite.T(), 1, seriesCount, "series count didn't match expected")
+}
+
+func (suite *testQuerySuite) TestClientWindowedAggregationWindowSmallerThanStep() {
+	adapter, err := tsdb.NewV3ioAdapter(suite.v3ioConfig, nil, nil)
+	if err != nil {
+		suite.T().Fatalf("failed to create v3io adapter. reason: %s", err)
+	}
+
+	labels1 := utils.LabelsFromStringList("os", "linux")
+	numberOfEvents := 10
+
+	var ingestedData []tsdbtest.DataPoint
+
+	for i := 0; i < numberOfEvents; i++ {
+		ingestedData = append(ingestedData, tsdbtest.DataPoint{Time: suite.basicQueryTime + int64(i)*tsdbtest.MinuteInMillis, Value: 10 * float64(i)})
+	}
+
+	testParams := tsdbtest.NewTestParams(suite.T(),
```

Comment:
Can you reuse all these "before" test blocks? Why to copy&paste if you can externalise   it?

Reply:
every test has different input data and expected data, so the overhead of trying to move it all to a generic "setup" function wouldn't be that "clean" i think. but worth thinking about it in another PR 😄

Reply:
ok, remember that big brother is watching...